Improve field documentation (#836)
Yelp/nrtsearch: docs/field_types/id.rst

ID Field
========

> This field serves as the primary term for the document. All document are required to have a value for this field. When updating a document, the _ID field is used to atomically replace the previous document with the new version. It is highly recommended for all indies to have an _ID field.

Reviewer: Typo
```suggestion
This field serves as the primary term for the document. All documents are required to have a value for this field. When updating a document, the _ID field is used to atomically replace the previous document with the new version. It is highly recommended for all indices to have an _ID field.
```

Reviewer: Also slightly confusing - all documents are _required_ to have a value for this field, but it's _recommended_ for all indices to have an _ID field. IMO we can just say that this is required, since it's not possible to update documents without _ID field.
```suggestion
This field serves as the primary term for the document. All indices are required to have an _ID field, and all documents are required to have a value for this field. When updating a document, the _ID field is used to atomically replace the previous document with the new version.
```
